Java/tests/src/test/java/com/microsoft/recognizers/text/tests/AbstractTest.java

```diff
@@ -36,9 +38,70 @@ public AbstractTest(TestCase currentCase) {
         this.currentCase = currentCase;
     }
 
+    private static Map<String, Integer> testCounter;
+    private static Map<String, Integer> passCounter;
+    private static Map<String, Integer> failCounter;
+    private static Map<String, Integer> skipCounter;
+
+    @BeforeClass
+    public static void before(){
+        testCounter = new LinkedHashMap<>();
+        passCounter = new LinkedHashMap<>();
+        failCounter = new LinkedHashMap<>();
+        skipCounter = new LinkedHashMap<>();
+    }
+
+    @AfterClass
+    public static void after(){
+        Map<String, String> counter = new LinkedHashMap<>();
+        for (Map.Entry<String, Integer> entry : testCounter.entrySet()) {
+            counter.put(entry.getKey(), entry.getValue().toString());
+        }
+        for (Map.Entry<String, String> entry : counter.entrySet()) {
+            Integer passValue = passCounter.getOrDefault(entry.getKey(), 0);
+            Integer failValue = failCounter.getOrDefault(entry.getKey(), 0);
+            Integer skipValue = skipCounter.getOrDefault(entry.getKey(), 0);
+            counter.put(entry.getKey(), "|\t" + entry.getValue() + "\t|\t" + passValue.toString() + "\t|\t" + skipValue.toString() + "\t|\t" + failValue.toString());
+        }
+        print(counter);
+    }
+
+    private static void print(Map<String, String> map) {
+        System.out.println("|\tTOTAL\t|\tPassed\t|\tSkipped\t|\tFailed\t||| Key");
+        for (Map.Entry<String, String> entry : map.entrySet()) {
+            System.out.println(entry.getValue() + "\t||| " + entry.getKey());
+        }
+    }
+
+    private void count(TestCase testCase) {
+        String key = testCase.recognizerName + "-" + testCase.language + "-" + testCase.modelName;
+        Integer current = testCounter.getOrDefault(key, 0);
+        testCounter.put(key, current + 1);
+    }
+
+    private void countPass(TestCase testCase) {
+        String key = testCase.recognizerName + "-" + testCase.language + "-" + testCase.modelName;
+        Integer current = passCounter.getOrDefault(key, 0);
+        passCounter.put(key, current + 1);
+    }
+
+    private void countSkip(TestCase testCase) {
+        String key = testCase.recognizerName + "-" + testCase.language + "-" + testCase.modelName;
+        Integer current = skipCounter.getOrDefault(key, 0);
+        skipCounter.put(key, current + 1);
+    }
+
+    private void countFail(TestCase testCase) {
```

Comment:
I see in the build log that it now counts many Fails. ex:
"|	TOTAL	|	Passed	|	Skipped	|	Failed	||| Key"
"|	41            |	0	        |	0	        |	41	        ||| DateTime-Chinese-DateExtractor"

But in the summary it says "Failures: 0" and the build passes. Is there some part of the new implementation missing?